www-apps/gitea/gitea-1.10.0.ebuild

```diff
@@ -0,0 +1,99 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+inherit golang-vcs-snapshot tmpfiles systemd
+
+EGO_PN="code.gitea.io/gitea"
+
+DESCRIPTION="A painless self-hosted Git service"
+HOMEPAGE="https://gitea.io"
+SRC_URI="https://github.com/go-gitea/gitea/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="Apache-2.0 BSD BSD-2 ISC MIT MPL-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64"
+IUSE="+acct pam sqlite"
+
+BDEPEND="dev-lang/go"
+COMMON_DEPEND="
+	acct? (
+		acct-group/git
+		acct-user/git[gitea]
+	)
+	pam? ( sys-libs/pam )"
+DEPEND="${COMMON_DEPEND}"
+RDEPEND="${COMMON_DEPEND}
+	dev-vcs/git"
+
+DOCS=( custom/conf/app.ini.sample CONTRIBUTING.md README.md )
+S="${WORKDIR}/${P}/src/${EGO_PN}"
+
+PATCHES=( "${FILESDIR}/gitea-logflags.patch" )
+
+gitea_make() {
+	local my_tags=(
+		bindata
+		$(usev pam)
+		$(usex sqlite 'sqlite sqlite_unlock_notify' '')
+	)
+	local my_makeopt=(
+		DRONE_TAG=${PV}
+		TAGS="${my_tags[@]}"
+		LDFLAGS="-extldflags \"${LDFLAGS}\""
+	)
+	GOPATH=${WORKDIR}/${P}:$(get_golibdir_gopath) emake "${my_makeopt[@]}" "$@"
+}
+
+src_prepare() {
+	default
+
+	local sedcmds=(
```

Comment:
Why not make this huge list into a patch? Can some of these be upstreamed, I see some are Gentoo-specific?

Reply:
with the removal of the non-needed seds (APP_ID and TRUSTED_FACETS (since 1.8.0) i don't think there is much to upstream, the path stuff is gentoo specific (does EPREFIX expansion even work in a patch?) and so is the logging, HTTP_ADDR used to be localhost via LOCAL_ROOT_URL (i *think*?) and that was pre-1.0 so i don't think that's what upstream wants either. RUN_MODE will be set to prod when the webinstaller ran successfully so no reason to upstream that either

Reply:
>(does EPREFIX expansion even work in a patch?) 

There's a syntax for it that works in a patch, I don't remember it now. `$(EPREFIX)` or `@EPREFIX@` I suppose.